### ISSUE TITLE
Support for  silhouette_samples() functionalities like sklearn.

### ISSUE THREAD

**Issue by srimantacse** [tslearn-team/tslearn#451]
Is there any support for silhouette_samples() functionalities as sklearn has.

**Comment by rtavenar** [comment]
Hi,

We don't have support for `silhouette_samples` at the moment.

In terms of implementation, given the current implementation of `silhouette_score` (https://github.com/tslearn-team/tslearn/blob/main/tslearn/clustering/utils.py#L66) that heavily relies on `sklearn`'s implem, it should be pretty much the same for one willing to adapt `silhouette_samples`:

1. compute cross-similarity metrics between points
2. rely on sklearn's implementation using `metric="precomputed"`

Anyone is welcome to open a PR on the topic.

**Comment by srimantacse** [comment]
@rtavenar thanks for the help. Able to get the sample score as you mentioned the steps. Thanks again.

**Comment by srimantacse** [comment]
@rtavenar please share me the process, I will do it.
Request to give me some time considering my schedule. 
Share me the process.

**Comment by rtavenar** [comment]
You can find info there: https://github.com/tslearn-team/tslearn/blob/main/CONTRIBUTING.md#more-details-on-pull-requests

No problem if it takes some time, of course, it's always great to welcome new contributors
